// @zoom-image/core/dist/utils.js
'use strict';

function s(e,n,r){return Math.max(n,Math.min(r,e))}function o(e){e.preventDefault();}var l=new Set(["ArrowUp","ArrowRight","ArrowDown","ArrowLeft"]);function u(e){if(l.has(e.key))return o(e),!1}var t;function c(){t=new AbortController;let{signal:e}=t;window.addEventListener("DOMMouseScroll",o,{signal:e}),window.addEventListener("wheel",o,{passive:!1,signal:e}),window.addEventListener("touchmove",o,{passive:!1,signal:e}),window.addEventListener("keydown",u,{signal:e});}function m(){t?.abort();}function f(e){if(!e)throw new Error("Please specify a container for the zoom image");let n=e.querySelector("img");if(!n)throw new Error("Please place an image inside the container");return n}function w(e,n){return {x:(e.x+n.x)/2,y:(e.y+n.y)/2}}function p(e,n){return r=>{e()&&n(r);}}var d=({domainStart:e,domainStop:n,rangeStart:r,rangeStop:i})=>a=>r+(i-r)*((a-e)/(n-e));

exports.clamp = s;
exports.disableScroll = c;
exports.enableScroll = m;
exports.getPointersCenter = w;
exports.getSourceImage = f;
exports.makeMaybeCallFunction = p;
exports.scaleLinear = d;
